skimage/measure/_label.py

```diff
@@ -38,6 +38,10 @@ def label(input, neighbors=None, background=None, return_num=False,
         as a neighbor.
         Accepted values are ranging from  1 to input.ndim. If ``None``, a full
         connectivity of ``input.ndim`` is used.
+    ratio : int, optional
```

Comment:
@sciunto Ok I can try to re-work it - how does this sound?
> Rather than connecting only pixels with equal values,  connect pixels that have a similar value, within the ratio specified by this argument. For example, if R is the ratio, a is the value of one pixel, and b is the value of one of its neighbors, those two pixels will be considered connected if a/R <= b <= a*R.

Note that this is sort of a generalization of the current approach, which could just be considered R=1. 
a/1 <= b <= a*1 simplifies to a = b, i.e. check for the same value.
(Technically my implementation uses < rather than <=, so perhaps I should change that)